### PR TITLE
Allow shared sockets config to be a sequence instead of a mapping

### PR DESCRIPTION
**Pull request by emulbreh**

### CODE REVIEW

File: lymph/config.py [hunk 1]
```diff
@@ -84,14 +87,21 @@ def get_instance(self, key, default_class=None, **kwargs):
...
-        return values[path[-1]]
+                if default is not Undefined:
+                    return default
+                raise KeyError()
+        try:
+            return values[path[-1]]
+        except KeyError:
+            if default is not Undefined:
+                return default
+            raise KeyError()
```
Comment (mouadino) [inline lymph/config.py]: maybe you want empty ``raise`` to reraise the above KeyError so that we can also see the key

File: lymph/config.py [hunk 2]
```diff
@@ -84,14 +87,21 @@ def get_instance(self, key, default_class=None, **kwargs):
...
         path = key.split('.')
         values = self.values
         for bit in path[:-1]:
             values = values[bit]
             if values is None:
-                raise KeyError
-        return values[path[-1]]
+                if default is not Undefined:
+                    return default
+                raise KeyError()
```
Comment (mouadino) [inline lymph/config.py]: ``raise KeyError(bit)`` ?

File: lymph/config.py (hunk 1 above)
Comment (emulbreh) [inline lymph/config.py]: I don't want to reraise, but we could add the full key to the error message.

**Comment by mouadino** [comment]
LGTM

**Comment by mamachanko** [comment]
example config?

**Pull request merged**